spec/requests/new_document_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+RSpec.describe "New Document" do
+  describe "GET /documents/choose-document-type" do
+    it "shows supertype when a supertype managed by content publisher is selected" do
+      supertype = Supertype.all.reject(&:managed_elsewhere).first
+      get choose_document_type_path, params: { supertype: supertype.id }
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to include(supertype.label)
+    end
+
+    it "redirects when a supertype managed elsewhere is selected" do
+      supertype = Supertype.all.find(&:managed_elsewhere)
+      get choose_document_type_path, params: { supertype: supertype.id }
+      expect(response).to redirect_to(supertype.managed_elsewhere_url)
+    end
+
+    it "returns an unprocessable response with an issue when a super type isn't selected" do
+      get choose_document_type_path
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.body)
+        .to include(I18n.t!("requirements.supertype.not_selected.form_message"))
```

Comment:
Is there a reason this expectation and https://github.com/alphagov/content-publisher/pull/1598/files#diff-a85fa69ab372a0c3eb3625fa606ceb90R56 do not use `have_selector` but the other expectations testing requirements messages do, like here - https://github.com/alphagov/content-publisher/pull/1598/files#diff-3463045769d93e0dafbeddae41166202R26?

Reply:
Yes, mostly we don't want to care about selectors. The ones that use have_selector are used to test that the message is displayed in an error context. We anticipate that the message would normally be displayed but just not in an error context.

The alternative approach to this would be to test that the flash was set for example `expect(flash["tried_to_publish"]).to be(true)` but this has the downside of not asserting behaviour.

Reply:
Ah okay that's good to know! 

However the two expectations I've mentioned above are both testing the flash error message contained within a `.gem-c-error-summary` element (as both are asserting requirements checks) so shouldn't they also be testing the selector like the other example?

Reply:
Ah sure I think I get you.

So in [this new document one](https://github.com/alphagov/content-publisher/pull/1598/files#diff-a85fa69ab372a0c3eb3625fa606ceb90R52-R57) we're testing that given an unprocessable request we show an issue. We're not too concerned about the context as our controller is not concerned about that.

Whereas in [this `.gem-c-error-summary` one](https://github.com/alphagov/content-publisher/pull/1598/files#diff-3463045769d93e0dafbeddae41166202R19-R28) the effect we are testing is not whether the redirected page shows the issue (as it will anyway) but that it shows it in a particular context. Thus the way to test that is to use the selector.

So testing logic is such that if we can avoid needing to use the selector then that is preferable - however that isn't the case for places where the controller is adjusting the context in which the error is presented.

Reply:
Oh I get you, thanks for the explanation!!